### PR TITLE
[Improvement] start|stop script files improvements

### PR DESCRIPTION
**Pull request by wolfboys**
## Further comments

start|stop script files improvements:

1) compatible with the case where the script is a soft link (reference apache tomcat)
2) optimize script readability

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: bin/stop_be.sh [hunk 1]
```diff
@@ -16,14 +16,26 @@
...
+done
+
+PRGDIR=`dirname "$PRG"`
+
+export DORIS_HOME=`cd "$PRG_DIR/.." >/dev/null; pwd`
+export PID_DIR=`cd "$PRGDIR" >/dev/null; pwd`
 
 signum=9
-if [ "x"$1 = "x--grace" ]; then
+if [ x"$1" != x"--grace" ]; then
```
Comment (compiletheworld) [inline bin/stop_be.sh]: This change may break the original semantic, which seems not correct.

File: bin/start_be.sh [hunk 2]
```diff
@@ -16,11 +16,31 @@
...
+  if expr "$link" : '/.*' > /dev/null; then
+    PRG="$link"
+  else
+    PRG=`dirname "$PRG"`/"$link"
+  fi
+done
+
+PRGDIR=`dirname "$PRG"`
+
+export DORIS_HOME=`cd "$PRG_DIR/.." >/dev/null; pwd`
```
Comment (compiletheworld) [inline bin/start_be.sh:34]: Where is `PRG_DIR` defined and what's the different between `PRG_DIR` and `PRGDIR`?

File: bin/start_be.sh [hunk 3]
```diff
@@ -16,11 +16,31 @@
...
+export PID_DIR=`cd "$PRGDIR" >/dev/null; pwd`
```
Comment (compiletheworld) [inline bin/start_be.sh:35]: Is `cd` necessary to evaluate `PID_DIR`? `cd` changes the current directory, which may lead to some subtle bugs.

File: bin/start_be.sh [hunk 4]
```diff
@@ -16,11 +16,31 @@
...
 
-curdir=$(dirname "$0")
-curdir=$(
-    cd "$curdir"
-    pwd
-)
+# resolve links - $0 may be a softlink
+PRG="$0"
+
+while [ -h "$PRG" ] ; do
```
Comment (compiletheworld) [inline bin/start_be.sh:22]: Consider command `readlink` to resolve the soft link problem?

**Comment by compiletheworld** [comment]
The changes of this PR seem not robust and not well tested, please resolve the comments and resubmit, thank you.

File: bin/start_be.sh (hunk 4 above)
Comment (wolfboys) [reply bin/start_be.sh:22]: yes. this method has been used in my project for a long time, you can see: https://github.com/apache/tomcat/blob/main/bin/startup.sh

File: bin/start_be.sh (hunk 3 above)
Comment (wolfboys) [reply bin/start_be.sh:35]: oh, sorry. it's my fault

**Review by wolfboys — COMMENTED** [review]
@compiletheworld PTAL

**Comment by compiletheworld** [comment]
LGTM

**Comment by wolfboys** [comment]
@compiletheworld PTAL

**Comment by wolfboys** [comment]
@compiletheworld @morningman PTAL

**Comment by yiguolei** [comment]
please rebase master and submit again.

**Comment by morningman** [comment]
@wolfboys

**Comment by wolfboys** [comment]
see #11325

**Comment by wolfboys** [comment]
> @wolfboys

ok

**Pull request closed without merging**